src/util/LazyJsonParser.cpp

```diff
@@ -218,40 +204,52 @@ void LazyJsonParser::parseAfterArrayPath(size_t& idx, size_t& materializeEnd) {
         break;
     }
   }
+  return std::nullopt;
 }
 
 // ____________________________________________________________________________
-std::string LazyJsonParser::constructResultFromParsedChunk(
+std::optional<nlohmann::json> LazyJsonParser::constructResultFromParsedChunk(
     size_t materializeEnd) {
-  std::string res;
-  if (materializeEnd == 0) {
-    return res;
+  size_t nextChunkStart =
+      materializeEnd == 0 ? 0 : std::min(materializeEnd + 1, input_.size());
+  if (input_.size() - nextChunkStart >= 1'000'000) {
+    throw std::runtime_error("Ill formed Json.");
+  }
+  if (nextChunkStart == 0) {
+    return std::nullopt;
   }
 
-  // Don't yield before arrayPath.
-  AD_CORRECTNESS_CHECK(!std::holds_alternative<BeforeArrayPath>(state_));
-
+  std::string resStr;
   if (yieldCount_ > 0) {
-    res = prefixInArray_;
+    resStr = prefixInArray_;
```

Comment:
Can use a ternary instead of three lines: `std::string resStr = yeldCount_ >0 ? prefixInArray_ : "";`